tests/client/amo/testApi.js

```diff
@@ -92,5 +92,134 @@ describe('amo.api', () => {
         mockApi.verify();
       });
     });
+
+    it('does not patch the version for existing reviews', () => {
+      const params = {
+        ...baseParams,
+        reviewId: 987,
+        body: 'some new body',
+        versionId: 99876,
+        addonSlug: 'chill-out',
+      };
+
+      mockApi
+        .expects('callApi')
+        .withArgs({
+          endpoint: `addons/addon/${params.addonSlug}/reviews/${params.reviewId}`,
+          body: {
+            // Make sure that version is not passed in.
+            ...defaultParams, body: params.body, version: undefined,
+          },
+          method: 'PATCH',
+          auth: true,
+          state: params.apiState,
+        })
+        .returns(Promise.resolve(genericApiResponse));
+
+      return submitReview(params).then(() => {
+        mockApi.verify();
+      });
+    });
+  });
+
+  describe('getUserReviews', () => {
+    it('gets all user reviews', () => {
+      const userId = 8877;
+      const reviewList = [fakeReview];
+      const response = { results: reviewList };
+      mockApi
+        .expects('callApi')
+        .withArgs({
+          endpoint: `accounts/account/${userId}/reviews`,
+        })
+        .returns(Promise.resolve(response));
+
+      return getUserReviews({ userId })
+        .then((reviews) => {
+          mockApi.verify();
+          assert.deepEqual(reviews, reviewList);
+        });
+    });
+
+    it('requires a user ID', () => {
+      mockApi.expects('callApi').never();
+      return getUserReviews()
+        .then(() => {
+          throw new Error('unexpected success');
+        })
+        .catch((error) => {
+          assert.equal(error.message, 'userId cannot be falsey');
+          mockApi.verify();
+        });
+    });
+
+    it('does not support paging yet', () => {
+      mockApi
+        .expects('callApi')
+        .returns(Promise.resolve({
+          results: [],
+          next: '/reviews/next-page/',
+        }));
+
+      return getUserReviews({ userId: 123 })
+        .then(() => {
+          throw new Error('unexpected success');
+        })
+        .catch((error) => {
```

Comment:
It probably makes sense to add a helper. I'm not sure what web-ext has would work. It looks like using that you could still get tests passing when they should fail.

```js
const mock = sinon.mock(something, 'someProp').expects('foo');
return doSomething()
  .then(makeSureItFails())
  .catch(() => {
    // Oops, error is "This test unexpectedly succeeded without an error"
    mock.verify();
   });
```

It looks like [chai-as-promised](https://github.com/domenic/chai-as-promised#assert-interface) has something for this:

```js
assert.isRejected(doSomething(), /userId cannot be falsey/);
// Looks like this would work, too:
assert.isRejected(doSomething(), /userId cannot be falsey/)
  .then(() => mock.verify());